SCClassLibrary/Common/Audio/InOut.sc

```diff
@@ -208,6 +208,9 @@ InFeedback : AbstractIn {
 		^this.multiNew('audio', numChannels, bus)
 	}
 	init { arg numChannels ... argBus;
+		numChannels.isNumber.or {
```

Comment:
>This should be an if clause, shouldn't it:

I believe `numChannels.isNumber.or {Error ...}` *does* work (even if it is a little eccentric! ) because `or` only evaluates the second arg if the first is false. But because it is hard to read I too would prefer something like

`numChannels.isNumber.not.if { ...`

> ... I'm not not shure what happens if a value 2.5 or -2.5 is given...

Good point: Negative numbers give a sensible Error `"wrong number of channels" - positive floats give a less helpful Error "Could not write def"

I think the check should indeed require a positive integer.

Reply:
Not blocking, because it's a _recommendation_, just note the preferred style in the [style guidelines wiki](https://github.com/supercollider/supercollider/wiki/Style-Guidelines:-SuperCollider#control-structures-trailing-closure-syntax).

Reply:
Interesting, but after looking up `False:-or` I see how using this method makes sense here. And indeeed `numChannels` _must_ be an integer (a float, even if it's a whole-numbered value throws an error...).

Reply:
>  I too would prefer something like `numChannels.isNumber.not.if { ...`

Agree, that is more idiomatic/familiar for this codebase 👍 

> I think the check should indeed require a positive integer

OK, so the guard against negative values is in `MultiOutUGen.initOutputs` https://github.com/supercollider/supercollider/blob/c21ed5b16c5b379f3c47fd1a45c93b28b2c8d822/SCClassLibrary/Common/Audio/UGen.sc#L579-L581 

It would seem like a guard against float values would make sense here as well ...

(from @cdbzb above)

> should we not fix this for all the subclasses of AbstractIn ?

After digging around, I'm suspecting that not only the float validation but indeed the numeric requirement needs to happen in `MultiOutUGen.initOutputs`, as they will all throw the "Non Boolean in test" error (the `>` operator in the code above is what chokes). 

I've looked at the subclasses in the standard library and certainly classes like `PlayBuf` are not happy if you have varying `numChannels`, and most of the rest don't even offer the parameter, hardcoding the value (e.g. `Pan2`).

Of course, this is a bigger change, and a LOT of extension classes extend `MultiOutUGen`, so this needs some vetting and approval!

Reply:
(sorry my previous comments posted late, I needed to "submit" the review ugh)

I'd propose the positive integer check [above](https://github.com/supercollider/supercollider/pull/6190#discussion_r1449372730), but I agree many UGens call `initOutputs` so it would be good to thoroughly check the implications.

Could you post the code that produces the "non-boolean in test" error? I'm only able to cause other errors with invalid inputs:
```supercollider
(
SynthDef("help-InFeedback", { arg out=0, in=0;
	var input, sound;
	input = InFeedback.ar(in, 1.1); // ERROR: SynthDef: could not write def: UGen: could not write def: BinaryOpFailureError
	// input = InFeedback.ar(in, \symb); // ERROR: Primitive '_BasicNew' failed. Index not an Integer
	sound = SinOsc.ar(input * 1300 + 300, 0, 0.4);
	Out.ar(out, sound);
}).add;
)
```

Reply:
> Could you post the code that produces the "non-boolean in test" error? 

Sure, as documented, any of the `In*` ugens don't accept a parameterized `numChannels`, giving that error which motivated this PR because it's so hard to understand :) . Relevant discussion on scsynth: https://scsynth.org/t/weird-error-non-boolean-in-test-with-infeedback-constructor/8772/1

```
(
add(SynthDef(\weird, { arg in, out, numChannels=2;
	var a = InFeedback.ar(in,numChannels);
	Out.ar(out,a);
	Out.ar(in,a);
}));
)
```

Reply:
Here's a non-`In*` ugen blowing up:

```
(
add(SynthDef(\weird, { arg buf, numChannels = 2;
	Out.ar(0,PlayBuf.ar(numChannels,buf));
}));
)
```

Reply:
in that example an Error message like "Playbuf: numchannels must be a positive Integer" wouldn't quite explain the problem to users who often don't get that numchannels has been implicitly set to a Control. A simple solution would be along the lines of `"cant set numchannels to %, needs to be a positive Integer".fmt(numchannels)` but perhaps we might actually check to see if numchannels is set to a Control and give an even more informative error?